application/src/command.rs

```diff
@@ -22,12 +22,11 @@ impl TryInto<TodoListMessage> for &Command {
     fn try_into(self) -> Result<TodoListMessage, Self::Error> {
         Ok(match self {
             Command::AddTask { name } => TodoListMessage::AddTask(
-                name.clone()
-                    .try_into()
+                name.try_into()
                     .map_err(|_| CommandError::TaskNameCannotBeEmpty)?,
```

Comment:
**Suggestion:** Consider using `if let` or `match` for error handling instead of `map_err` to provide more specific error messages. [enhancement]
```suggestion
                name.try_into().map_err(|err| {
                    match err {
                        SomeSpecificError => CommandError::SpecificErrorDetail,
                        _ => CommandError::TaskNameCannotBeEmpty
                    }
                })?,
```